java/vector/src/main/codegen/templates/FixedValueVectors.java

```diff
@@ -39,20 +39,26 @@
  * ${minor.class} implements a vector of fixed width values.  Elements in the vector are accessed
  * by position, starting from the logical start of the vector.  Values should be pushed onto the
  * vector sequentially, but may be randomly accessed.
+<#if (type.width > 0) >
  *   The width of each element is ${type.width} byte(s)
+</#if>
  *   The equivalent Java primitive is '${minor.javaType!type.javaType}'
  *
  * NB: this class is automatically generated from ${.template_name} and ValueVectorTypes.tdd using FreeMarker.
  */
 public final class ${className} extends BaseDataValueVector implements FixedWidthVector{
   private static final org.slf4j.Logger logger = org.slf4j.LoggerFactory.getLogger(${className}.class);
 
+  <#if (type.width < 0)>
+  public final int TYPE_WIDTH;
+  <#else>
   public static final int TYPE_WIDTH = ${type.width};
```

Comment:
I am not sure about having `TYPE_WIDTH` defined as static in some classes and non static in others. 

I think it's cleaner if we define `TYPE_WIDTH` only as static and define another class method `getTypeWidth()`

Reply:
Another reason is all caps are usually used to define static constant. Type width for fixed binary is not constant. so I think it's cleaner to not reuse the same variable for fixed binary.